packages/components/src/internal/components/chart/ChartBuilderModal.tsx

```diff
@@ -482,35 +526,40 @@ const ChartPreview: FC<ChartPreviewProps> = memo(props => {
 
         setLoadingData(true);
 
-        LABKEY_VIS.GenericChartHelper.queryChartData(divId, queryConfig_, measureStore => {
-            const rowCount = LABKEY_VIS.GenericChartHelper.getMeasureStoreRecords(measureStore).length;
-            const _previewMsg = getChartRenderMsg(chartConfig, rowCount, true);
-
-            if (rowCount > MAX_POINT_DISPLAY) {
-                if (chartConfig.renderType === 'box_plot') {
-                    chartConfig.pointType = 'outliers';
-                    chartConfig.geomOptions.boxFillColor = BLUE_HEX_COLOR;
-                } else if (chartConfig.renderType === 'line_plot') {
-                    chartConfig.geomOptions.hideDataPoints = true;
+        LABKEY_VIS.GenericChartHelper.queryChartData(
+            divId,
+            queryConfig_,
+            chartConfig,
+            (measureStore, trendlineData) => {
+                const rowCount = LABKEY_VIS.GenericChartHelper.getMeasureStoreRecords(measureStore).length;
+                const _previewMsg = getChartRenderMsg(chartConfig, rowCount, true);
+
+                if (rowCount > MAX_POINT_DISPLAY) {
+                    if (chartConfig.renderType === 'box_plot') {
+                        chartConfig.pointType = 'outliers';
+                        chartConfig.geomOptions.boxFillColor = BLUE_HEX_COLOR;
+                    } else if (chartConfig.renderType === 'line_plot') {
+                        chartConfig.geomOptions.hideDataPoints = true;
+                    }
                 }
-            }
 
-            // adjust height, width, and marginTop for the chart config for the preview, but not to save with the chart
-            var chartConfig_ = {
-                ...chartConfig,
-                height: 350,
-                width,
-            };
-            if (!savedChartModel || savedChartModel.visualizationConfig.chartConfig.geomOptions.marginTop === 20) {
-                chartConfig_.geomOptions.marginTop = 15;
-            }
+                // adjust height, width, and marginTop for the chart config for the preview, but not to save with the chart
+                var chartConfig_ = {
+                    ...chartConfig,
+                    height: 350,
+                    width,
+                };
+                if (!savedChartModel || savedChartModel.visualizationConfig.chartConfig.geomOptions.marginTop === 20) {
+                    chartConfig_.geomOptions.marginTop = 15;
+                }
 
-            if (ref?.current) ref.current.innerHTML = ''; // clear again, right before render
-            LABKEY_VIS.GenericChartHelper.generateChartSVG(divId, chartConfig_, measureStore);
+                if (ref?.current) ref.current.innerHTML = ''; // clear again, right before render
```

Comment:
I don't think we need to be defensive here.

```suggestion
                if (ref.current) ref.current.innerHTML = ''; // clear again, right before render
```

Reply:
done